server/src/main/java/org/apache/gravitino/server/web/rest/TableOperations.java

```diff
@@ -84,6 +89,16 @@ public Response listTables(
           () -> {
             Namespace tableNS = NamespaceUtil.ofTable(metalake, catalog, schema);
             NameIdentifier[] idents = dispatcher.listTables(tableNS);
+            idents =
+                MetadataFilterHelper.filterByExpression(
+                    metalake,
+                    "((ANY(USE_CATALOG,METALAKE,CATALOG,SCHEMA)) && "
+                        + "(ANY(USE_SCHEMA,METALAKE,CATALOG,SCHEMA)) &&"
+                        + " ( ANY(SELECT_TABLE,METALAKE,CATALOG,SCHEMA,TABLE) || "
+                        + " ANY(MODIFY_TABLE,METALAKE,CATALOG,SCHEMA,TABLE) || TABLE::OWNER)) ||"
+                        + "ANY(OWNER,METALAKE,CATALOG,SCHEMA)",
```

Comment:
`SCHEMA` owner should have `USE_CATATLOG` privilege first.

Reply:
fixed